authz/client_test.go

```diff
@@ -14,6 +14,118 @@ import (
+			group:            "dashboard.grafana.app",
+			resource:         "dashboards",
+			verb:             "list",
+			want:             true,
+		},
+		{
+			name:             "Permission matches group/resource/name",
+			tokenPermissions: []string{"dashboard.grafana.app/dashboards/dashUID:get"},
+			group:            "dashboard.grafana.app",
+			resource:         "dashboards",
+			verb:             "get",
+			resourceName:     "dashUID",
+			want:             true,
+		},
+		{
+			name:             "Permission does not match verb",
+			tokenPermissions: []string{"dashboard.grafana.app/dashboards:list"},
+			group:            "dashboard.grafana.app",
+			resource:         "dashboards",
+			verb:             "get",
+			want:             false,
+		},
+		{
+			name:             "Permission matches wildcard verb",
+			tokenPermissions: []string{"dashboard.grafana.app/dashboards:*"},
+			group:            "dashboard.grafana.app",
+			resource:         "dashboards",
+			verb:             "get",
+			want:             true,
+		},
+		{
+			name:             "Permission does not match group/resource/name",
+			tokenPermissions: []string{"dashboard.grafana.app/dashboards/otherDashUID:get"},
+			group:            "dashboard.grafana.app",
+			resource:         "dashboards",
+			verb:             "get",
+			resourceName:     "dashUID",
+			want:             false,
+		},
+		{
+			name:             "Invalid permission missing verb",
+			tokenPermissions: []string{"dashboard.grafana.app/dashboards"},
+			group:            "dashboard.grafana.app",
+			resource:         "dashboards",
+			verb:             "list",
+			want:             false,
+		},
+		{
+			name:             "Permission on the wrong group",
+			tokenPermissions: []string{"other-group.grafana.app/dashboards:list"},
+			group:            "dashboard.grafana.app",
+			resource:         "dashboards",
+			verb:             "list",
+			want:             false,
+		},
+		{
+			name:             "Permission on the wrong resource",
+			tokenPermissions: []string{"dashboard.grafana.app/other-resource:list"},
+			group:            "dashboard.grafana.app",
+			resource:         "dashboards",
+			verb:             "list",
+			want:             false,
+		},
+		{
+			name:             "Permission without group are skipped",
+			tokenPermissions: []string{":get"},
+			group:            "dashboard.grafana.app",
+			resource:         "dashboards",
+			verb:             "get",
+			resourceName:     "dashUID",
+			want:             false,
+		},
+		{
+			name:             "Group permission work",
+			tokenPermissions: []string{"dashboard.grafana.app:list"},
+			group:            "dashboard.grafana.app",
+			resource:         "dashboards",
+			verb:             "list",
+			want:             true,
+		},
+		{
+			name:             "Wildcard action works",
+			tokenPermissions: []string{"dashboard.grafana.app/dashboard:*"},
+			group:            "dashboard.grafana.app",
+			resource:         "dashboards",
```

Comment:
I'm lacking quite a bit of context here, but is this the intended behavior? IMO it seems dangerous to implement something like this. For instance, if we were to ever introduce another resource called `dashboard-annotations` (as a crude example), the token would automatically have access over it.

```suggestion
			tokenPermissions: []string{"dashboard.grafana.app/dashboards:*"},
			group:            "dashboard.grafana.app",
			resource:         "dashboards",
```

Reply:
Excellent point thanks! I'll fix that tomorrow